### PR TITLE
fix: initialize v_cache to avoid NaNs

### PR DESCRIPTION
**Pull request by Atry**
This PR fixes https://github.com/vllm-project/vllm/issues/641 by initializing v_cache.

The root cause is that the padding of v_cache will be used to dot-multiply by the padding of QK logits. Normally the dot product of v_cache padding and QK logits padding should be 0, because QK logits padding value is 0, therefore the result on the padded tokens should not contribute to the final result. However, when v_cache padding is NaN, the result would be NaN unexpectedly, because `0 * nan` is `nan`.

With the help of this PR, `v_cache` padding would become a finite value to avoid NaN result.

### CODE REVIEW

**Comment by Atry** [comment]
This PR is exported from our internal commit, which is already tested working. I am going to merge it directly since we don't have a public CI.
